Makefile.am

```diff
@@ -225,53 +244,70 @@ endif
 	rm -f $(DESTDIR)$(sysconfdir)/opencryptoki/opencryptoki.conf || true
 	rm -f $(DESTDIR)$(sysconfdir)/opencryptoki/strength.conf || true
 
-
 if ENABLE_TESTCASES
 PKCS11_SO_PIN ?= 76543210
 PKCS11_USER_PIN ?= 01234567
 PKCS11_VHSM_PIN ?= 1234567890
+if AIX
+PKCSLIB = '@libdir@/opencryptoki/libopencryptoki.a(libopencryptoki.so.0)'
+else
+PKCSLIB ?= @libdir@/opencryptoki/libopencryptoki.so
+endif
+
 
 ci-prepare:
-	killall -HUP pkcsslotd || true
+if AIX
+	chmod u+x $(KILLALL)
+endif
+	$(KILLALL) -HUP pkcsslotd || true
 	${srcdir}/testcases/ciconfig.sh "$(sysconfdir)/opencryptoki" "$(sysconfdir)/opencryptoki" "$(sysconfdir)/opencryptoki"
 	@sbindir@/pkcsslotd
 	@sbindir@/pkcsstats --reset-all
-	for slot in `awk '/slot (.*)/ { print $$2; }' $(sysconfdir)/opencryptoki/opencryptoki.conf`; do @sbindir@/pkcsconf -c $$slot -t | grep "Flags:" | grep -q TOKEN_INITIALIZED || PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so ${srcdir}/testcases/init_token.sh $$slot; done
-#	cd ${srcdir}/testcases && PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCS11_VHSM_PIN=$(PKCS11_VHSM_PIN) PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so ./init_vhsm.exp 42
+	for slot in `awk '/^slot (.*)/ { print $$2; }' $(sysconfdir)/opencryptoki/opencryptoki.conf`; do @sbindir@/pkcsconf -c $$slot -t | grep "Flags:" | grep -q TOKEN_INITIALIZED || PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCSLIB=$(PKCSLIB) ${srcdir}/testcases/init_token.sh $$slot; done
+#	cd ${srcdir}/testcases && PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCS11_VHSM_PIN=$(PKCS11_VHSM_PIN) PKCSLIB=$(PKCSLIB) ./init_vhsm.exp 42
 #	echo "VHSM_MODE" >> "$(sysconfdir)/opencryptoki/ep11tok42.conf"
 
-installcheck-local: all
-	killall -HUP pkcsslotd || true
+installcheck-local: all check
+if AIX
+	chmod u+x $(KILLALL)
+endif
+	$(KILLALL) -HUP pkcsslotd || true
 	@sbindir@/pkcsslotd
 	if test ! -z ${PKCS11_TEST_USER}; then				\
 		chmod 777 ${srcdir}/testcases &&			\
 		cd ${srcdir}/testcases &&                               \
-		su ${PKCS11_TEST_USER} -s /bin/bash -c "PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so sh ./ock_tests.sh || true"; \
+		su ${PKCS11_TEST_USER} -s /bin/bash -c "PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCSLIB=$(PKCSLIB) sh ./ock_tests.sh || true"; \
 	else								\
 		cd ${srcdir}/testcases && 				\
-		PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so sh ./ock_tests.sh || true; \
+		PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCSLIB=$(PKCSLIB) ./ock_tests.sh || true; \
 	fi
-	killall -HUP pkcsslotd
+	$(KILLALL) -HUP pkcsslotd
 
 ci-installcheck: ci-prepare installcheck
-	killall -HUP pkcsslotd || true
+if AIX
+	chmod u+x $(KILLALL)
```

Comment:
why is this required? Can't the kill-all script have the executable flags set right away?